src/Raven.Server/Documents/Queries/Graph/QueryQueryStep.cs

```diff
@@ -40,6 +41,18 @@ public class QueryQueryStep : IGraphQueryStep
             _context = documentsContext;
             _resultEtag = existingResultEtag;
             _token = token;
+
+            if (!string.IsNullOrEmpty(queryMetadata.CollectionName)) //not a '_' collection
```

Comment:
I'm probably missing something but how this can filter out '_' collection as stated in the comment?

Reply:
That would result in "index / all / collection" options that we'll always need to check.
I don't really like that